rest/data/auth.go

```diff
@@ -0,0 +1,91 @@
+package data
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/evergreen-ci/cedar/util"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/jpillora/backoff"
+	"github.com/pkg/errors"
+)
+
+/////////////////////////////
+// DBConnector Implementation
+/////////////////////////////
+
+// EvergreenProxyAuthLogRead sends a http request to Evergreen to check if the
+// given user is authorized to read the log belonging to the given project. If
+// the user is authorized, nil is returned, otherwise a gimlet.Responder with
+// the appropriate http error code is returned.
+func (dbc *DBConnector) EvergreenProxyAuthLogRead(ctx context.Context, userToken, baseURL, resourceId string) gimlet.Responder {
+	urlString := fmt.Sprintf("%s?resource=%s&resource_type=project&permission=project_logs&required_level=10", baseURL, resourceId)
+	req, err := http.NewRequest(http.MethodGet, urlString, nil)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "error creating http request"))
+	}
+	req = req.WithContext(ctx)
+	req.Header = map[string][]string{"mci-token": {userToken}}
+
+	client := util.GetHTTPClient()
+	defer util.PutHTTPClient(client)
+
+	b := &backoff.Backoff{
+		Min:    100 * time.Millisecond,
+		Max:    10 * time.Second,
+		Factor: 2,
+	}
+	var responder gimlet.Responder
+	for i := 0; i < 10; i++ {
+		resp, err := client.Do(req)
```

Comment:
do you have to reset the request if it fails? I'm not sure. We have a function object wrapper that does the retry logic in evergreen so I'm not sure.

Reply:
Requests are safe to use: "Or we just say it's only safe to reuse a Request if Body is nil or the the Request.Body has been read to EOF. (the caller could wrap the Body in their own wrapper which notes EOF if they wanted)." from Brad Fitz here https://github.com/golang/go/issues/19653.